### ISSUE TITLE
Model failed with error: UnsupportedDTypeForOp(BF16, "matmul")

### ISSUE THREAD

**Issue by Xiao-Chenguang** [EricLBuehler/mistral.rs#437]
**Describe the bug**
The model raise a UnsupportedDTypeForOp(BF16, "matmul") error what ever I type in interactive mode.

**Latest commit**
Current head is `d66a8f4cacc7ca276c408a0515326b9f1d3aaeff`

The error log is here.
```
01:55:38  |llm|usrname@server mistral.rs ±|master ✗|→ ./target/release/mistralrs-server -i plain -m ~/xxx/llamaModel -a phi3
2024-06-15T12:57:24.955207Z  INFO mistralrs_server: avx: true, neon: false, simd128: false, f16c: true
2024-06-15T12:57:24.955303Z  INFO mistralrs_server: Sampling method: penalties -> temperature -> topk -> topp -> multinomial
2024-06-15T12:57:24.955337Z  INFO mistralrs_server: Model kind is: normal (no quant, no adapters)
2024-06-15T12:57:24.956128Z  INFO mistralrs_core::pipeline::normal: Loading `tokenizer.json` at `/server/homes/c/usrname/xxx/llamaModel`
2024-06-15T12:57:25.081610Z  INFO mistralrs_core::pipeline::normal: Loading `"tokenizer.json"` locally at `"/server/homes/c/usrname/xxx/llamaModel/tokenizer.json"`
2024-06-15T12:57:25.081648Z  INFO mistralrs_core::pipeline::normal: Loading `config.json` at `/server/homes/c/usrname/xxx/llamaModel`
2024-06-15T12:57:25.202296Z  INFO mistralrs_core::pipeline::normal: Loading `"config.json"` locally at `"/server/homes/c/usrname/xxx/llamaModel/config.json"`
2024-06-15T12:57:25.665632Z  INFO mistralrs_core::pipeline::paths: Loading `"/server/homes/c/usrname/xxx/llamaModel/model-00002-of-00002.safetensors"` locally at `"/server/homes/c/usrname/xxx/llamaModel/model-00002-of-00002.safetensors"`
2024-06-15T12:57:25.776066Z  INFO mistralrs_core::pipeline::paths: Loading `"/server/homes/c/usrname/xxx/llamaModel/model-00001-of-00002.safetensors"` locally at `"/server/homes/c/usrname/xxx/llamaModel/model-00001-of-00002.safetensors"`
2024-06-15T12:57:26.207803Z  INFO mistralrs_core::pipeline::normal: Loading `tokenizer_config.json` at `/server/homes/c/usrname/xxx/llamaModel`
2024-06-15T12:57:26.746719Z  INFO mistralrs_core::pipeline::normal: Loading `"tokenizer_config.json"` locally at `"/server/homes/c/usrname/xxx/llamaModel/tokenizer_config.json"`
2024-06-15T12:57:26.746982Z  INFO mistralrs_core::utils::normal: DType selected is BF16.
2024-06-15T12:57:26.747008Z  INFO mistralrs_core::pipeline::normal: Loading model `/server/homes/c/usrname/xxx/llamaModel` on Cpu...
2024-06-15T12:57:26.747169Z  INFO mistralrs_core::pipeline::normal: Model config: Config { vocab_size: 32064, hidden_act: Silu, hidden_size: 3072, intermediate_size: 8192, num_hidden_layers: 32, num_attention_heads: 32, num_key_value_heads: 32, rms_norm_eps: 1e-5, rope_theta: 10000.0, bos_token_id: Some(1), eos_token_id: Some(32000), rope_scaling: None, max_position_embeddings: 4096, use_flash_attn: false, sliding_window: Some(2047), original_max_position_embeddings: 4096 }
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 67/67 [00:01<00:00, 46.18it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 128/128 [00:03<00:00, 63.52it/s]
2024-06-15T12:57:30.111308Z  INFO mistralrs_core::pipeline::chat_template: bos_toks = "<s>", eos_toks = "<|endoftext|>", unk_tok = <unk>
2024-06-15T12:57:30.113087Z  INFO mistralrs_server: Model loaded.
2024-06-15T12:57:30.113194Z  INFO mistralrs_server::interactive_mode: Starting interactive loop with sampling params: SamplingParams { temperature: Some(0.1), top_k: Some(32), top_p: Some(0.1), top_n_logprobs: 0, frequency_penalty: Some(0.1), presence_penalty: Some(0.1), stop_toks: None, max_len: Some(4096), logits_bias: None, n_choices: 1 }
> how are you
2024-06-15T12:57:43.615748Z ERROR mistralrs_core::engine: prompt step - Model failed with error: UnsupportedDTypeForOp(BF16, "matmul")
2024-06-15T12:57:43.615980Z ERROR mistralrs_server::interactive_mode: Got a model error: "unsupported dtype BF16 for op matmul", response: ChatCompletionResponse { id: "0", choices: [Choice { finish_reason: "error", index: 0, message: ResponseMessage { content: "", role: "assistant" }, logprobs: None }], created: 1718456263, model: "/server/homes/c/usrname/xxx/llamaModel", system_fingerprint: "local", object: "chat.completion", usage: Usage { completion_tokens: 0, prompt_tokens: 8, total_tokens: 8, avg_tok_per_sec: 421.05264, avg_prompt_tok_per_sec: inf, avg_compl_tok_per_sec: NaN, total_time_sec: 0.019, total_prompt_time_sec: 0.0, total_completion_time_sec: 0.0 } }
```

**Comment by EricLBuehler** [comment]
Hi @Xiao-Chenguang! Thank you for reporting this. I just merged what should be a fix in #438, can you please `git pull` and try it again?

**Comment by Xiao-Chenguang** [comment]
> Hi @Xiao-Chenguang! Thank you for reporting this. I just merged what should be a fix in #438, can you please `git pull` and try it again?

Thank you @EricLBuehler. I confirm your fix works for this issue.

**Comment by EricLBuehler** [comment]
I'll close this issue given that it is fixed, please feel free to reopen!
